test/05-updates.test.js

```diff
@@ -285,54 +289,53 @@ test('Pear.updates should notify Platform stage, Platform release updates (diffe
 
   comment('restaging rig platform')
   const staging = rig.local.stage({ channel: `test-${rig.id}`, name: `test-${rig.id}`, dir: rig.artefactDir, dryRun: false, bare: true })
+  teardown(() => Helper.teardownStream(staging))
   await Helper.pick(staging, { tag: 'final' })
   comment('rig platform restaged')
   comment('waiting for update')
-  const update1 = await update1ActualPromise
-  const update1Version = update1?.value?.version
+  const update1 = await untilUpdate1
+  const update1Version = update1?.version
   const pearUpdateLength = update1Version.length
   ok(pearUpdateLength > pearVersionLength, `platform version.length incremented (v${update1Version?.fork}.${update1Version?.length})`)
 
   comment('releasing rig platform')
-  const update2Promise = await update2LazyPromise
-  const update2ActualPromise = running.inspector.awaitPromise(update2Promise.objectId)
   const releasing = rig.local.release({ channel: `test-${rig.id}`, name: `test-${rig.id}`, dir: rig.artefactDir })
+  teardown(() => Helper.teardownStream(releasing))
   await Helper.pick(releasing, { tag: 'final' })
 
   comment('waiting for platform update notification')
-  const update2 = await update2ActualPromise
-  const update2Version = update2?.value?.version
+  const update2 = await untilUpdate2
+  const update2Version = update2?.version
   const pearUpdate2Key = update2Version.key
   const pearUpdate2Length = update2Version.length
 
   is(pearUpdate2Key, rig.key, 'platform release update matches staging key')
   ok(pearUpdate2Length > pearUpdateLength, `platform version length incremented (v${update2Version?.fork}.${update2Version?.length})`)
 
-  await running.inspector.evaluate('__PEAR_TEST__.close()')
-  await running.inspector.close()
-  const { code } = await running.until.exit
-  is(code, 0, 'exit code is 0')
+  pipe.write('exit')
```

Comment:
can we try `pipe.end()` for parent?

then in the child worker, we can `pipe.on('end', cb)` or `pipe.on('close', cb)` with a `cb` to updates.end()